In [1]:
pip install scipy

Defaulting to user installation because normal site-packages is not writeable
Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 24.3.1 -> 25.0.1
[notice] To update, run: C:\Users\Lenovo\AppData\Local\Microsoft\WindowsApps\PythonSoftwareFoundation.Python.3.13_qbz5n2kfra8p0\python.exe -m pip install --upgrade pip


<h1> DATA ANALYTICS PROJECT </h1>

<h2> United Kingdom Road Accident Data Analysis</h2>
<h3> Inclusive years : 2019 - 2022</h3>

<h1> IMPORTING LIBRARIES</h1>

In [2]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import warnings
import scipy
warnings.filterwarnings('ignore')
from scipy.stats import f_oneway

<h1>  DATA PREPARATION </h1>

In [3]:
accident = pd.read_csv("dataset\\accident_data.csv")

In [4]:
accident

Index Accident_Severity Accident Date   Latitude  \
0       200701BS64157           Serious    05/06/2019  51.506187   
1       200701BS65737           Serious    02/07/2019  51.495029   
2       200701BS66127           Serious    26/08/2019  51.517715   
3       200701BS66128           Serious    16/08/2019  51.495478   
4       200701BS66837            Slight    03/09/2019  51.488576   
...               ...               ...           ...        ...   
660674  201091NM01760            Slight    18/02/2022  57.374005   
660675  201091NM01881            Slight    21/02/2022  57.232273   
660676  201091NM01935            Slight    23/02/2022  57.585044   
660677  201091NM01964           Serious    23/02/2022  57.214898   
660678  201091NM02142           Serious    28/02/2022  57.575210   

                   Light_Conditions           District Area  Longitude  \
0             Darkness - lights lit  Kensington and Chelsea  -0.209082   
1                          Daylight  Kensington and Chelsea  -0.173647   
2       Darkness - lighting unknown  Kensington and Chelsea  -0.210215   
3                          Daylight  Kensington and Chelsea  -0.202731   
4             Darkness - lights lit  Kensington and Chelsea  -0.192487   
...                             ...                     ...        ...   
660674                     Daylight                Highland  -3.467828   
660675       Darkness - no lighting                Highland  -3.809281   
660676                     Daylight                Highland  -3.862727   
660677       Darkness - no lighting                Highland  -3.823997   
660678                     Daylight                Highland  -3.895673   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
0                          1                   2                     Dry   
1                          1                   2             Wet or damp   
2                          1                   3                     Dry   
3                          1                   4                     Dry   
4                          1                   2                     Dry   
...                      ...                 ...                     ...   
660674                     2                   1                     Dry   
660675                     1                   1            Frost or ice   
660676                     1                   3            Frost or ice   
660677                     1                   2             Wet or damp   
660678                     1                   1             Wet or damp   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
0       Single carriageway               Urban     Fine no high winds   
1       Single carriageway               Urban  Raining no high winds   
2                      NaN               Urban                    NaN   
3       Single carriageway               Urban     Fine no high winds   
4                      NaN               Urban                    NaN   
...                    ...                 ...                    ...   
660674  Single carriageway               Rural     Fine no high winds   
660675  Single carriageway               Rural     Fine no high winds   
660676  Single carriageway               Rural     Fine no high winds   
660677  Single carriageway               Rural     Fine no high winds   
660678    Dual carriageway               Rural  Snowing no high winds   

                                Vehicle_Type  
0                                        Car  
1                                        Car  
2                      Taxi/Private hire car  
3       Bus or coach (17 or more pass seats)  
4                              Other vehicle  
...                                      ...  
660674                                   Car  
660675                                   Car  
660676                                   Car  
660677                 Motorcycle over 500cc  
660

In [5]:
accident.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 660679 entries, 0 to 660678
Data columns (total 14 columns):
 #   Column                   Non-Null Count   Dtype  
---  ------                   --------------   -----  
 0   Index                    660679 non-null  object 
 1   Accident_Severity        660679 non-null  object 
 2   Accident Date            660679 non-null  object 
 3   Latitude                 660654 non-null  float64
 4   Light_Conditions         660679 non-null  object 
 5   District Area            660679 non-null  object 
 6   Longitude                660653 non-null  float64
 7   Number_of_Casualties     660679 non-null  int64  
 8   Number_of_Vehicles       660679 non-null  int64  
 9   Road_Surface_Conditions  659953 non-null  object 
 10  Road_Type                656159 non-null  object 
 11  Urban_or_Rural_Area      660664 non-null  object 
 12  Weather_Conditions       646551 non-null  object 
 13  Vehicle_Type             660679 non-null  object 
dtypes: f

In [6]:
accident.describe()

Latitude      Longitude  Number_of_Casualties  Number_of_Vehicles
count  660654.000000  660653.000000         660679.000000       660679.000000
mean       52.553866      -1.431210              1.357040            1.831255
std         1.406922       1.383330              0.824847            0.715269
min        49.914430      -7.516225              1.000000            1.000000
25%        51.490690      -2.332291              1.000000            1.000000
50%        52.315641      -1.411667              1.000000            2.000000
75%        53.453452      -0.232869              1.000000            2.000000
max        60.757544       1.762010             68.000000           32.000000

In [7]:
accident.dtypes

Index                       object
Accident_Severity           object
Accident Date               object
Latitude                   float64
Light_Conditions            object
District Area               object
Longitude                  float64
Number_of_Casualties         int64
Number_of_Vehicles           int64
Road_Surface_Conditions     object
Road_Type                   object
Urban_or_Rural_Area         object
Weather_Conditions          object
Vehicle_Type                object
dtype: object

<h2> Converting Object to Datetime Datatype</h2>

In [8]:
accident['Accident Date'] = pd.to_datetime(accident['Accident Date'], dayfirst = True, errors = 'coerce')

In [9]:
accident['Year'] = accident['Accident Date'].dt.year
accident['Month'] = accident['Accident Date'].dt.month
accident['Day'] = accident['Accident Date'].dt.day
accident['DayofWeek'] = accident['Accident Date'].dt.dayofweek

In [10]:
accident.dtypes

Index                              object
Accident_Severity                  object
Accident Date              datetime64[ns]
Latitude                          float64
Light_Conditions                   object
District Area                      object
Longitude                         float64
Number_of_Casualties                int64
Number_of_Vehicles                  int64
Road_Surface_Conditions            object
Road_Type                          object
Urban_or_Rural_Area                object
Weather_Conditions                 object
Vehicle_Type                       object
Year                                int32
Month                               int32
Day                                 int32
DayofWeek                           int32
dtype: object

<h3> Clearing null values </h3>

In [11]:
accident.isnull().sum()

Index                          0
Accident_Severity              0
Accident Date                  0
Latitude                      25
Light_Conditions               0
District Area                  0
Longitude                     26
Number_of_Casualties           0
Number_of_Vehicles             0
Road_Surface_Conditions      726
Road_Type                   4520
Urban_or_Rural_Area           15
Weather_Conditions         14128
Vehicle_Type                   0
Year                           0
Month                          0
Day                            0
DayofWeek                      0
dtype: int64

In [12]:
accident['Latitude'] = accident['Latitude'].fillna(accident['Latitude'].mode()[0])
accident['Longitude'] = accident['Longitude'].fillna(accident['Longitude'].mode()[0])
accident['Road_Surface_Conditions'] = accident['Road_Surface_Conditions'].fillna('unknown surface conditions')
accident['Road_Type'] = accident['Road_Type'].fillna('unaccounted')
accident['Weather_Conditions'] = accident['Weather_Conditions'].fillna('unaccounted weather conditions')
accident['Urban_or_Rural_Area'] = accident['Urban_or_Rural_Area'].fillna(accident['Urban_or_Rural_Area'].mode()[0])

In [13]:
 accident.isnull().sum()

Index                      0
Accident_Severity          0
Accident Date              0
Latitude                   0
Light_Conditions           0
District Area              0
Longitude                  0
Number_of_Casualties       0
Number_of_Vehicles         0
Road_Surface_Conditions    0
Road_Type                  0
Urban_or_Rural_Area        0
Weather_Conditions         0
Vehicle_Type               0
Year                       0
Month                      0
Day                        0
DayofWeek                  0
dtype: int64

<h3> Changing category types </h3>

In [14]:
accident['Accident_Severity'] = accident['Accident_Severity'].astype('category')
accident['Latitude'] = accident['Latitude'].astype('category')
accident['Longitude'] = accident['Longitude'].astype('category')
accident['Light_Conditions'] = accident['Light_Conditions'].astype('category')
accident['District Area'] = accident['District Area'].astype('category')
accident['Road_Surface_Conditions'] = accident['Road_Surface_Conditions'].astype('category')
accident['Road_Type'] = accident['Road_Type'].astype('category')
accident['Urban_or_Rural_Area'] = accident['Urban_or_Rural_Area'].astype('category')
accident['Weather_Conditions'] = accident['Weather_Conditions'].astype('category')
accident['Vehicle_Type'] = accident['Vehicle_Type'].astype('category')

In [15]:
accident.dtypes

Index                              object
Accident_Severity                category
Accident Date              datetime64[ns]
Latitude                         category
Light_Conditions                 category
District Area                    category
Longitude                        category
Number_of_Casualties                int64
Number_of_Vehicles                  int64
Road_Surface_Conditions          category
Road_Type                        category
Urban_or_Rural_Area              category
Weather_Conditions               category
Vehicle_Type                     category
Year                                int32
Month                               int32
Day                                 int32
DayofWeek                           int32
dtype: object

<h1> UNIVARIATE</h1>

<h2>INSIGHT 1</h2>
<h3> Which road type helds more accidents?</h3>

In [16]:
accident['Road_Type'].count()

np.int64(660679)

In [17]:
accident['Road_Type'].value_counts()

Road_Type
Single carriageway    492143
Dual carriageway       99424
Roundabout             43992
One way street         13559
Slip road               7041
unaccounted             4520
Name: count, dtype: int64

<p><b>INSIGHT: Single Carriageways have the most accidents, making up over 75% pf total case records </b></p>

<h2>INSIGHT 2</h2>
<h3>What is the most commong weather condition when accidents happen? </h3>

In [18]:
accident['Weather_Conditions'].value_counts()

Weather_Conditions
Fine no high winds                520885
Raining no high winds              79696
Other                              17150
unaccounted weather conditions     14128
Raining + high winds                9615
Fine + high winds                   8554
Snowing no high winds               6238
Fog or mist                         3528
Snowing + high winds                 885
Name: count, dtype: int64

<p><b>INSIGHT: Fine with no High Winds weather condition accounts for 79% of accidents, suggesting driver behavior plays a significant role rather than extreme weather</b></p>

<h2> INSIGHT 3 </h2>
<h3> What's the Most Frequent Light Condition during accidents?</h3>

In [19]:
accident['Light_Conditions'].value_counts()

Light_Conditions
Daylight                       484880
Darkness - lights lit          129335
Darkness - no lighting          37437
Darkness - lighting unknown      6484
Darkness - lights unlit          2543
Name: count, dtype: int64

<p><b>INSIGHT: Daylight accidents (73%) most commonly occur rather than nightime accidents</b></p>

<h2> INSIGHT 4 </h2>
<h3> Which vehicle type is involved in the most accidents? </h3>

In [20]:
accident['Vehicle_Type'].value_counts()

Vehicle_Type
Car                                      497992
Van / Goods 3.5 tonnes mgw or under       34160
Bus or coach (17 or more pass seats)      25878
Motorcycle over 500cc                     25657
Goods 7.5 tonnes mgw and over             17307
Motorcycle 125cc and under                15269
Taxi/Private hire car                     13294
Motorcycle over 125cc and up to 500cc      7656
Motorcycle 50cc and under                  7603
Goods over 3.5t. and under 7.5t            6096
Other vehicle                              5637
Minibus (8 - 16 passenger seats)           1976
Agricultural vehicle                       1947
Pedal cycle                                 197
Data missing or out of range                  6
Ridden horse                                  4
Name: count, dtype: int64

<p><b>INSIGHT: Cars are involved in the most accidents resulting to over 75% of the overall accident records </b></p>

<h2> INSIGHT 5 </h2>
<h3>Which district has the highest number of accidents?</h3>

In [21]:
accident['District Area'].value_counts()

District Area
Birmingham            13491
Leeds                  8898
Manchester             6720
Bradford               6212
Sheffield              5710
                      ...  
Berwick-upon-Tweed      153
Teesdale                142
Shetland Islands        133
Orkney Islands          117
Clackmannanshire         91
Name: count, Length: 422, dtype: int64

<p><b>INSIGHT: among all given districs within United Kingdom, Birmingham District helds about 2% of the overall accident records among different districts </b></p>

<h1>BIVARIATE</h1>

<h2> INSIGHT 6 </h2>
<h3> In what road type does most of the vehicles go?</h3>

In [22]:
accident.groupby('Road_Type')['Number_of_Vehicles'].sum().sort_values(ascending=False)

Road_Type
Single carriageway    883845
Dual carriageway      199819
Roundabout             83643
One way street         21497
Slip road              13501
unaccounted             7567
Name: Number_of_Vehicles, dtype: int64

<h4> INSIGHTS: Based from the results, SIngle Carriageway is the go to road type of most of the drivers recorded</h4>

<h2> INSIGHT 7</h2>
<h3> How fatal are the accidents within rural and urban areas?</h3>

In [23]:
urban_severity = accident[accident['Urban_or_Rural_Area'] == 'Urban']['Accident_Severity'].value_counts()

In [24]:
rural_severity = accident[accident['Urban_or_Rural_Area'] == 'Rural']['Accident_Severity'].value_counts()

In [25]:
urban_severity

Accident_Severity
Slight     367714
Serious     50904
Fatal        3060
Name: count, dtype: int64

In [26]:
rural_severity

Accident_Severity
Slight     196077
Serious     37312
Fatal        5601
Name: count, dtype: int64

<h4> INSIGHT: Ironically, Most fatal accidents occur within rural areas where there are lesser cars than the urban areas. However it is proven the slight accidents can and might happen regardless of the area</h4>

<h2> INSIGHT 8</h2>
<h3> Accident severity counts within different weather conditions</h3>

In [27]:
weather_severity = accident.groupby('Weather_Conditions')['Accident_Severity'].value_counts().unstack().fillna(0)
#severe_weather = weather_severity['Severe'].sort_values(ascending=False)

In [28]:
weather_severity

Accident_Severity               Fatal  Serious  Slight
Weather_Conditions                                    
Fine + high winds                 175     1245    7134
Fine no high winds               7100    72046  441739
Fog or mist                        82      483    2963
Other                             165     1801   15184
Raining + high winds              145     1261    8209
Raining no high winds             848     9468   69380
Snowing + high winds                3      109     773
Snowing no high winds              36      565    5637
unaccounted weather conditions    107     1239   12782

<h4> Among all recorded data, Fine with no High winds weather conditions accounts to most of the accidents. This suggests that other factor might be the reason why such accidents happen.</h4>

<h2> INSIGHT 9 </h2>
<h3> Number of Vehicles catered in different road surface conditions</h3>

In [29]:
vehicles_by_surface = accident.groupby('Road_Surface_Conditions')['Number_of_Vehicles'].sum().sort_values(ascending=False)

In [30]:
vehicles_by_surface.sum() -824793

np.int64(385079)

<h4> INSIGHTS: From the results we can infer that, most drivers prefer to use dry roads(68%) rather than other road surface conditions where the rest of the other drivers drove.<h4>

<h2> INSIGHT 10 </h2>
<h3> How light conditions accounts to different accident severities</h3>

In [31]:
light_severity = accident.groupby('Light_Conditions')['Accident_Severity'].value_counts().unstack().fillna(0)

In [32]:
light_severity

Accident_Severity            Fatal  Serious  Slight
Light_Conditions                                   
Darkness - lighting unknown     68      794    5622
Darkness - lights lit         1860    19130  108345
Darkness - lights unlit         45      360    2138
Darkness - no lighting        1612     7174   28651
Daylight                      5076    60759  419045

<h4> INSIGHTS: Based from the results, most of the given light conditions results to slight accident severities. However fatalities also does occur mostly on streets with broad daylight, in the dark with lights lit, and when there is no lighting.</h4>

<h1> MULTIVARIATE </h1>

<h2> INSIGHT 11</h2>
<h3> How serious accidents happen depending on light conditions, road type, and weather conditions?</h3>

In [33]:
accident[accident['Accident_Severity'] == 'Serious'].count()

Index                      88217
Accident_Severity          88217
Accident Date              88217
Latitude                   88217
Light_Conditions           88217
District Area              88217
Longitude                  88217
Number_of_Casualties       88217
Number_of_Vehicles         88217
Road_Surface_Conditions    88217
Road_Type                  88217
Urban_or_Rural_Area        88217
Weather_Conditions         88217
Vehicle_Type               88217
Year                       88217
Month                      88217
Day                        88217
DayofWeek                  88217
dtype: int64

In [34]:
severe_conditions = accident[accident['Accident_Severity'] == 'Serious'].groupby(['Light_Conditions', 'Road_Type', 'Weather_Conditions']).size().reset_index(name='Count')
top_severe = severe_conditions.sort_values('Count', ascending=False)

In [35]:
top_severe

Light_Conditions           Road_Type     Weather_Conditions  \
244                     Daylight  Single carriageway     Fine no high winds   
82         Darkness - lights lit  Single carriageway     Fine no high winds   
217                     Daylight    Dual carriageway     Fine no high winds   
248                     Daylight  Single carriageway  Raining no high winds   
190       Darkness - no lighting  Single carriageway     Fine no high winds   
..                           ...                 ...                    ...   
211       Darkness - no lighting         unaccounted   Raining + high winds   
231                     Daylight      One way street   Snowing + high winds   
25   Darkness - lighting unknown          Roundabout  Snowing no high winds   
258                     Daylight           Slip road   Snowing + high winds   
267                     Daylight         unaccounted   Snowing + high winds   

     Count  
244  41598  
82   11056  
217   6427  
248   4188  
190   4154  
..     ...  
211      0  
231      0  
25       0  
258      0  
267      0  

[270 rows x 4 columns]

<h4> INSIGHTS: Based from the current records most serious accidents happen during daylight in a single carriageway road with a fine with no high winds, such conditions accounts to 47% of all serious accident conditions recorded</h4>

<h2> INSIGHT 12</h2>
<h3> How fatal does then Road Type and its Surface Condition is?</h3>

In [36]:
surface_road_severity = accident.groupby(['Road_Surface_Conditions', 'Road_Type', 'Accident_Severity']).size().sort_values(ascending=False).reset_index(name='Count')

In [37]:
surface_road_severity

Road_Surface_Conditions           Road_Type Accident_Severity   Count
0                           Dry  Single carriageway            Slight  279489
1                   Wet or damp  Single carriageway            Slight  117798
2                           Dry    Dual carriageway            Slight   56889
3                           Dry  Single carriageway           Serious   48873
4                           Dry          Roundabout            Slight   27955
..                          ...                 ...               ...     ...
103  unknown surface conditions      One way street             Fatal       0
104  unknown surface conditions          Roundabout             Fatal       0
105  unknown surface conditions           Slip road             Fatal       0
106  unknown surface conditions           Slip road           Serious       0
107  unknown surface conditions         unaccounted             Fatal       0

[108 rows x 4 columns]

<h4> INSIGHTS: Within such results, Dry Single Carriageway road type accounts to most of the slight accident severity</h4>

<h2> INSIGHT 13</h2>
<h3> What light conditions in each district results to number of casualties</h3>

In [38]:
casualties_by_district_light = accident.groupby(['District Area', 'Light_Conditions'])['Number_of_Casualties'].count().sort_values(ascending=False).reset_index()

In [39]:
casualties_by_district_light

District Area             Light_Conditions  Number_of_Casualties
0                Birmingham                     Daylight                  9667
1                     Leeds                     Daylight                  6482
2                Manchester                     Daylight                  4610
3                  Bradford                     Daylight                  4459
4                 Sheffield                     Daylight                  4211
...                     ...                          ...                   ...
2105              Islington       Darkness - no lighting                     0
2106                 Exeter  Darkness - lighting unknown                     0
2107  Weymouth and Portland  Darkness - lighting unknown                     0
2108            Wyre Forest  Darkness - lighting unknown                     0
2109               Wychavon  Darkness - lighting unknown                     0

[2110 rows x 3 columns]

<h4> INSIGHTS: In Birmingham where the highest number of casualties are recorded, most of the accidents happen during broad daylight resulting to approximately 9667 number of casualties recorded.</h4>

<h2> INSIGHT 14</h2>

In [40]:
severe_by_district_road_vehicle = accident[accident['Accident_Severity'] == 'Serious'].groupby(['District Area', 'Road_Type', 'Vehicle_Type']).size().sort_values(ascending=False).reset_index(name='Count')

In [41]:
severe_by_district_road_vehicle

District Area           Road_Type                         Vehicle_Type  \
0        Birmingham  Single carriageway                                  Car   
1             Leeds  Single carriageway                                  Car   
2          Bradford  Single carriageway                                  Car   
3       Westminster  Single carriageway                                  Car   
4         Sheffield  Single carriageway                                  Car   
...             ...                 ...                                  ...   
40507          York         unaccounted                        Other vehicle   
40508          York         unaccounted                          Pedal cycle   
40509          York         unaccounted                         Ridden horse   
40510          York         unaccounted                Taxi/Private hire car   
40511          York         unaccounted  Van / Goods 3.5 tonnes mgw or under   

       Count  
0        727  
1        667  
2        530  
3        473  
4        438  
...      ...  
40507      0  
40508      0  
40509      0  
40510      0  
40511      0  

[40512 rows x 4 columns]

<h4> INSIGHTS: Based from the results about 727 count of all the recorded serious accidents happen with cars in a single carriageway road type within the Birmingham District</h4>

<h2> INSIGHT 15</h2>

In [42]:
severity_proportion = accident.groupby(['Road_Type', 'Weather_Conditions', 'Urban_or_Rural_Area'])['Accident_Severity'].value_counts().sort_values(ascending=False).unstack(fill_value=0).reset_index()

In [43]:
severity_proportion

Accident_Severity         Road_Type              Weather_Conditions  \
0                  Dual carriageway               Fine + high winds   
1                  Dual carriageway               Fine + high winds   
2                  Dual carriageway               Fine + high winds   
3                  Dual carriageway              Fine no high winds   
4                  Dual carriageway              Fine no high winds   
..                              ...                             ...   
157                     unaccounted           Snowing no high winds   
158                     unaccounted           Snowing no high winds   
159                     unaccounted  unaccounted weather conditions   
160                     unaccounted  unaccounted weather conditions   
161                     unaccounted  unaccounted weather conditions   

Accident_Severity Urban_or_Rural_Area  Fatal  Serious  Slight  
0                               Rural     17      115     742  
1                         Unallocated      0        0       0  
2                               Urban     15       65     565  
3                               Rural    956     4753   31309  
4                         Unallocated      0        0       0  
..                                ...    ...      ...     ...  
157                       Unallocated      0        0       0  
158                             Urban      1        1      13  
159                             Rural      3       13     119  
160                       Unallocated      0        0       0  
161                             Urban      1       54     712  

[162 rows x 6 columns]

<h4> INSIGHTS: the results shows how accident severity is distributed among rural and urban areas considering the road type and weather conditions.</h4>

<h2> INSIGT 16</h2>

In [44]:
monthly_severity = accident.groupby(['Year', 'Month', 'Accident_Severity']).size().reset_index(name='Count')

In [45]:
monthly_severity

Year  Month Accident_Severity  Count
0    2019      1             Fatal    239
1    2019      1           Serious   2009
2    2019      1            Slight  13107
3    2019      2             Fatal    173
4    2019      2           Serious   1639
..    ...    ...               ...    ...
139  2022     11           Serious   1631
140  2022     11            Slight  11858
141  2022     12             Fatal     90
142  2022     12           Serious   1071
143  2022     12            Slight   8464

[144 rows x 4 columns]

<h4> INSIGHT: The results show how many accidents happen within different months of each year in consideration with such accident severities</h4>

<h2> INSIGHT 17</h2>

In [46]:
day_severity = accident.groupby(['DayofWeek', 'Accident_Severity']).size().sort_values(ascending=False).reset_index(name='Count')

In [47]:
day_severity

DayofWeek Accident_Severity  Count
0           5            Slight  91852
1           3            Slight  85974
2           2            Slight  85957
3           4            Slight  84154
4           1            Slight  81527
5           6            Slight  74706
6           0            Slight  59631
7           5           Serious  14000
8           6           Serious  13074
9           4           Serious  12633
10          2           Serious  12488
11          3           Serious  12440
12          1           Serious  11918
13          0           Serious  11664
14          6             Fatal   1522
15          0             Fatal   1385
16          5             Fatal   1326
17          4             Fatal   1113
18          2             Fatal   1113
19          1             Fatal   1105
20          3             Fatal   1097

<h4> INSIGHTS: Based from the records most accidents happen during Fridays resulting to a count of 91852 number of slight accident severities</h4>

<h2> INSIGHT 18 </h2>

In [48]:
weather_month = accident.groupby(['Month', 'Weather_Conditions'])['Number_of_Casualties'].sum().sort_values(ascending=False).reset_index()

In [49]:
weather_month

Month    Weather_Conditions  Number_of_Casualties
0        6    Fine no high winds                 64819
1        5    Fine no high winds                 64372
2       10    Fine no high winds                 64131
3        9    Fine no high winds                 63584
4        7    Fine no high winds                 62678
..     ...                   ...                   ...
103      8  Snowing + high winds                     5
104      5  Snowing + high winds                     3
105      9  Snowing + high winds                     2
106      7  Snowing + high winds                     1
107      6  Snowing + high winds                     0

[108 rows x 3 columns]

<h4> INSIGHTS: In june where Fine with no High winds are usual, aboout 64819 number of casualties are affected duting different accidents</h4>

<h2> INSIGHT 19</h2>

In [50]:
 top_severe_months = accident[accident['Accident_Severity'] == 'Serious'].groupby(['Month']).size().reset_index(name='Count').sort_values('Count', ascending=False).head(3)

In [51]:
top_severe_months

Month  Count
8      9   7987
9     10   7939
5      6   7916

<h4> INSIGHTS: Most serious accidents happen during the month of September</h4>

<h2> INSIGHT 20</h2>

In [52]:
vehicles_trend = accident.groupby(['Year', 'Month', 'Accident_Severity'])['Number_of_Vehicles'].sum().sort_values(ascending=False).reset_index()

In [53]:
vehicles_trend

Year  Month Accident_Severity  Number_of_Vehicles
0    2019     11            Slight               26256
1    2019      5            Slight               25439
2    2019      7            Slight               25145
3    2021     11            Slight               25043
4    2020     10            Slight               24832
..    ...    ...               ...                 ...
139  2022      7             Fatal                 214
140  2022      4             Fatal                 209
141  2022      2             Fatal                 207
142  2022      1             Fatal                 180
143  2022     12             Fatal                 162

[144 rows x 4 columns]


<h4> INSIGHTS: Based from the given results we can infer that within the month of November during 2019, most of the slight accidents happen affecting the total of 26256 vehicles or 3% of all the road accidents recorded</h4>